### ISSUE TITLE
No debug data in DL4J Training UI (and no errors/exceptions), output NaN when testing

### ISSUE THREAD

**Issue by maksgraczyk** [deeplearning4j/deeplearning4j#2675]
I'm using the following model (adapted from one of the ComputerVision examples):
```
new NeuralNetConfiguration.Builder()
                        .seed(123)
                        .iterations(10)
                        .activation(Activation.RELU)
                        .weightInit(WeightInit.XAVIER)
                        .gradientNormalization(GradientNormalization.RenormalizeL2PerLayer)
                        .optimizationAlgo(OptimizationAlgorithm.STOCHASTIC_GRADIENT_DESCENT)
                        .learningRate(0.01)
                        .momentum(0.9)
                        .regularization(true)
                        .l2(1e-3)
                        .updater(Updater.ADAGRAD)
                        .useDropConnect(true)
                        .list()
                        .layer(0, new ConvolutionLayer.Builder(4, 4)
                                .name("cnn1")
                                .nIn(3)
                                .stride(1, 1)
                                .nOut(20)
                                .build())
                        .layer(1, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX, new int[]{2, 2})
                                .name("pool1")
                                .build())
                        .layer(2, new ConvolutionLayer.Builder(3, 3)
                                .name("cnn2")
                                .stride(1,1)
                                .nOut(40)
                                .build())
                        .layer(3, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX, new int[]{2, 2})
                                .name("pool2")
                                .build())
                        .layer(4, new ConvolutionLayer.Builder(3, 3)
                                .name("cnn3")
                                .stride(1,1)
                                .nOut(60)
                                .build())
                        .layer(5, new SubsamplingLayer.Builder(SubsamplingLayer.PoolingType.MAX, new int[]{2, 2})
                                .name("pool3")
                                .build())
                        .layer(6, new ConvolutionLayer.Builder(2, 2)
                                .name("cnn3")
                                .stride(1,1)
                                .nOut(80)
                                .build())
                        .layer(7, new DenseLayer.Builder()
                                .name("ffn1")
                                .nOut(160)
                                .dropOut(0.5)
                                .build())
                        .layer(8, new OutputLayer.Builder(LossFunctions.LossFunction.NEGATIVELOGLIKELIHOOD)
                                .nOut(1)
                                .activation(Activation.SOFTMAX)
                                .build())
                        .backprop(true).pretrain(false)
                        .setInputType(InputType.convolutional(150, 150, 3))
                        .build();
```
I've also set up a debugging GUI:
```
try {
            UIServer server = UIServer.getInstance();
            StatsStorage storage = new InMemoryStatsStorage();
            server.attach(storage);
            model.setListeners(new StatsListener(storage));
        }
        catch (Exception e) {
            e.printStackTrace();
        }
```
However, when I try to fit the model, nothing happens in the DL4J Training GUI overview section. I also don't get any errors or exceptions. When testing, I get the "NaN" results.

The fit function:
```
private void fit(BufferedImage image) throws Exception
    {
        ImageIO.write(image, "png", new File("tmp.png"));
        ImageRecordReader reader = new ImageRecordReader(150, 150, 3, new ParentPathLabelGenerator());
        reader.initialize(new FileSplit(new File("tmp.png")));
        DataSetIterator dataIter = new RecordReaderDataSetIterator(reader, 4, 0, 1);
        while (dataIter.hasNext())
        {
            DataNormalization normalization = new NormalizerMinMaxScaler();
            DataSet set = dataIter.next();
            normalization.fit(set);
            normalization.transform(set);
            MainGUI.model.fit(set);
        }
    }
```

The check function:
```
private void check(BufferedImage image) throws Exception
    {
        ImageIO.write(image, "png", new File("tmp.png"));
        ImageRecordReader reader = new ImageRecordReader(150, 150, 3, new ParentPathLabelGenerator());
        reader.initialize(new FileSplit(new File("tmp.png")));
        DataSetIterator dataIter = new RecordReaderDataSetIterator(reader, 4);
        while (dataIter.hasNext())
        {
            DataNormalization normalization = new NormalizerMinMaxScaler();
            DataSet set = dataIter.next();
            normalization.fit(set);
            normalization.transform(set);
            System.out.println(MainGUI.model.output(set.getFeatures(), false));
        }
    }
```

Removing the normalizing code from the check and fit functions doesn't work.

I don't know what's wrong. Could you help me with the problem?

**Comment by raver119** [comment]
Now we have OpProfiler that helps finding where NaNs come from.
